src/webgpu/shader/validation/uniformity/uniformity.spec.ts

```diff
@@ -354,3 +355,900 @@ fn main(@builtin(${t.params.builtin}) p : ${t.params.type}) {
 
     t.expectCompileResult(t.params.uniform, code);
   });
+
+function expectedUniformity(uniform: string, init: string): boolean {
```

Comment:
nit:
Could this be a const?
Something like this:
```
const expected_uniformity = {
  always: true;
  never: false;
  init: {
     none: true;
     uniform: true;
     nonuniform: false;
  };
};
```

Reply:
You can then use `expected_uniformity['always']` or `expected_uniformity[foo]`, where foo is a string

Reply:
Though I guess this would require a bit of switching logic in the test runner, i.e. `uniform === 'init' ? expected_uniformity['init'][init]: expected_uniformity[uniform];`